src/identity/interaction/email-password/credentials/CreateCredentialsHandler.ts

```diff
@@ -0,0 +1,59 @@
+import { randomBytes } from 'crypto';
+import type { Adapter } from 'oidc-provider';
+import { v4 } from 'uuid';
+import { BasicRepresentation } from '../../../../http/representation/BasicRepresentation';
+import type { Representation } from '../../../../http/representation/Representation';
+import { APPLICATION_JSON } from '../../../../util/ContentTypes';
+import { NotImplementedHttpError } from '../../../../util/errors/NotImplementedHttpError';
+import type { AdapterFactory } from '../../../storage/AdapterFactory';
+import type { AccountStore } from '../storage/AccountStore';
+import type { CredentialsHandlerInput } from './CredentialsHandler';
+import { CredentialsHandler } from './CredentialsHandler';
+import { CLIENT_CREDENTIALS_WEBID_KEY } from './CredentialsUtil';
+
+/**
+ * Handles the creation of credential tokens.
+ * Requires a `name` field in the input JSON body,
+ * that will be used to generate the ID token.
+ */
+export class CreateCredentialsHandler extends CredentialsHandler {
+  private readonly adapter: Adapter;
+  private readonly accountStore: AccountStore;
+
+  public constructor(factory: AdapterFactory, accountStore: AccountStore) {
+    super();
+    this.adapter = factory.createStorageAdapter('Client');
+    this.accountStore = accountStore;
+  }
+
+  public async canHandle({ body }: CredentialsHandlerInput): Promise<void> {
+    if (typeof body.name !== 'string') {
+      throw new NotImplementedHttpError();
+    }
+  }
+
+  public async handle({ operation, body: { webId, name }}: CredentialsHandlerInput): Promise<Representation> {
+    const settings = await this.accountStore.getSettings(webId);
+    const id = `${(name as string).replace(/\W/gu, '-')}_${v4()}`;
```

Comment:
Doesn't hurt to sanitize the whole thing (also #807)

Reply:
True, doesn't matter much either way. The only reason to sanitize here is to help the user out a bit actually since they have to form encode this data when sending it to our server, and this way we make it so the id is already a valid form encoded string. Internally we don't care since we just use it for strict string comparisons.